plugins/Makefile.am

```diff
@@ -120,11 +132,17 @@ EXTRA_DIST += ${BZIP2SRC} BZIP2_LICENSE
 if HAVE_LOCAL_BZ2
 lib__nch5bzip2_la_SOURCES += ${BZIP2SRC}
 endif
-lib__nch5bzip2_la_LDFLAGS = ${plugin_version_info}
-plugin_LTLIBRARIES += lib__nch5bzip2.la
+lib__nch5bzip2_la_LDFLAGS = ${plugin_version_info} $(AM_LDFLAGS)
+plugins_to_install += lib__nch5bzip2.la
 
 endif #ENABLE_FILTER_TESTING
 
+if ENABLE_PLUGIN_DIR
```

Comment:
Have you tested this on the platforms we support?

Reply:
I've tested this on Cygwin and MinGW [on my fork](https://github.com/DWesl/netcdf-c/tree/cygwin-plugin-build) and locally.

I think you've also tested this, since this is basically the solution proposed in #2440. 

I have a test for what this is trying to do (fix #2431) in the Cygwin CI (`make install DESTDIR=/tmp/pretend-root && test ! -d /tmp/pretend-root/$(pwd)`, more or less); I could try to add a similar test to the `installcheck` target, but that might take a while.